File: 21-merge-two-sorted-lists/21-merge-two-sorted-lists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode mergeTwoLists(ListNode list1, ListNode list2) {
        ListNode dummy= new ListNode(-1);
        ListNode ans=dummy;
        while(list1!=null && list2!=null){
            if(list1.val<list2.val){
                ans.next=new ListNode(list1.val);
                list1=list1.next;
            }
            else
                {
                    ans.next=new ListNode(list2.val);
                    list2=list2.next;
                }
            ans=ans.next;
        }
        while(list1!=null){
            ans.next=new ListNode(list1.val);
            list1=list1.next;
            ans=ans.next;
        }
            while(list2!=null){
                ans.next=new ListNode(list2.val);
                ans=ans.next;
                list2=list2.next;
                }
        return dummy.next;
    }
}

File: 21-merge-two-sorted-lists/21-merge-two-sorted-lists.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* mergeTwoLists(ListNode* l1, ListNode* l2) {
        ListNode* ans=new ListNode(-1);
        ListNode* temp=ans;
        while(l1!=NULL && l2!=NULL){
            if(l1->val<l2->val){
                ans->next=l1;
                ans=ans->next;
                l1=l1->next;
            }
            else
            {
                ans->next=l2;
                ans=ans->next;
                l2=l2->next;
            }
        }
        //if second list gets empty then we insert all remaining l1 element
        while(l1!=NULL){
            ans->next=l1;
            ans=ans->next;
            l1=l1->next;
        }
        while(l2!=NULL){
            ans->next=l2;
                ans=ans->next;
                l2=l2->next;
        }
        return temp->next;
    }
};